node/src/store.rs

```diff
@@ -1,9 +1,214 @@
+use celestia_types::ExtendedHeader;
+use dashmap::mapref::entry::Entry;
+use dashmap::DashMap;
+use std::sync::atomic::{AtomicU64, Ordering};
+use tendermint::Hash;
+use thiserror::Error;
+use tracing::{error, info, instrument};
+
+trait StoreTrait {
```

Comment:
I'm also wondering if that shouldn't be an async trait. web storage api looks synchronous, but we don't know what db are we gonna use there